dandischema/models.py

```diff
@@ -1191,17 +1197,25 @@ class PublishedAsset(Asset, Publishable):
     schemaKey = "Asset"
 
     @validator("digest")
-    def digest_bothhashes(cls, values):
-        try:
-            digest = values[DigestType.dandi_etag]
-            if "-" not in digest or len(digest.split("-")[0]) != 32:
+    def digest_allhashes(cls, values):
+        digest = values.get(DigestType.dandi_etag, None)
+        if digest is not None:
+            if not re.fullmatch(DandiETag.REGEX, digest):
                 raise ValueError("Digest is missing dandi-etag value")
-            digest = values[DigestType.sha2_256]
-            if len(digest) != 64:
+            digest = values.get(DigestType.sha2_256, None)
+            if digest is None or len(digest) != 64:
                 raise ValueError("Digest is missing sha2_256 value")
```

Comment:
again possibly a confusing message since might have a digest but which is not legit.  Might need to be split into two conditions.